### DISCUSSION TITLE
Installating DeePMD and DP-GEN in the same Conda environment

### DISCUSSION THREAD

**Discussion by RemiLacroix-IDRIS** [deepmodeling/deepmd-kit#2985] [Q&A]
Hello,

I am trying to install both DeePMD and DP-GEN in the same Conda environment but so far I have been struggling to make that work.

I tried to install from the official DeepModeling Conda channel but it seems the DP-GEN version available there is pretty old and I'm getting some conflicts.

I thought I would use the Conda-Forge channel but I'm not sure that the DeePMP build available there supports multi-GPU multi-nodes configuration as Horovod is not available.

Do you have any suggestions?

Best regards,
Rémi Lacroix

**Comment by njzjz** [comment]
If you don't mind mixing `conda` and `pip`, you can use `pip` directly.

> as Horovod is not available.

It's reasonable to have Horovod in the conda-forge.

**Reply by njzjz** [reply]
Horovod is now available in the conda-forge channel.

**Reply by RemiLacroix-IDRIS** [reply]
I tested `CONDA_OVERRIDE_CUDA="11.8" conda create -n deepmd-2.2.7 deepmd-kit==2.2.7=cuda118* lammps horovod dpgen -c conda-forge`, the installation works fine but we are facing some issues at runtime:
```
    File ".../envs/deepmd-2.2.7/lib/python3.11/site-packages/tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py", line 189, in __call__
Node: 'ExponentialDecay/Floor'
2 root error(s) found.
  (0) UNKNOWN: JIT compilation failed.
         [[{{node ExponentialDecay/Floor}}]]
         [[ExponentialDecay/_3]]
  (1) UNKNOWN: JIT compilation failed.
         [[{{node ExponentialDecay/Floor}}]]
0 successful operations.
0 derived errors ignored.
```
Any ideas?

**Reply by njzjz** [reply]
Might be the same issue as https://github.com/conda-forge/tensorflow-feedstock/issues/296. Did you manually enable the JIT, as shown in the error message?

**Reply by RemiLacroix-IDRIS** [reply]
That's indeed the same issue. If we enabled anything manually, we are not aware we did.

I have tried to apply the workarounds proposed in that issue but so far I am struggling...

**Reply by RemiLacroix-IDRIS** [reply]
> I have tried to apply the workarounds proposed in that issue but so far I am struggling...

I gave up on getting nvcc installed in the Conda env, I used our own CUDA 11.8 module and that allowed me to use one of the workaround.

**Reply by njzjz** [reply]
> If we enabled anything manually, we are not aware we did.

I know the reason. XLA is enabled by Horovod by default. See https://horovod.readthedocs.io/en/stable/xla.html

**Reply by RemiLacroix-IDRIS** [reply]
Ok.

Now we would like to be able to use the DeePMD plugin for Lammps but the version from "conda-forge" channel does not seem to be patched. Is the version from the "deepmodeling" channel patched?

What would you recommend in our specific case where we want to have DeePMD and DP-GEN installed in the same environment while also providing a patched version of LAMMPS?

**Reply by njzjz** [reply]
Both channels do not patch LAMMPS. The [plugin mechanism](https://docs.lammps.org/Developer_plugins.html) is used.

Several days ago I merged https://github.com/conda-forge/deepmd-kit-feedstock/pull/63 to fix the 
problem that the LAMMPS plugin environment variable was not set automatically. You may check if your version is no older than this fix.

**Reply by RemiLacroix-IDRIS** [reply]
Was the conda-forge channel updated already? My version does not seems to have the fix but when trying to update with `conda update -c conda-forge lammps`, it does not find any new versions bit it tries to downgrade my TensorFlow packages to CPU...

Here is the version currently installed:
> lammps                    2023.08.02      cpu_py311_hf0e8f25_openmpi_7    conda-forge

**Reply by njzjz** [reply]
deepmd-kit was updated, not lammps

**Reply by RemiLacroix-IDRIS** [reply]
Sorry, I misunderstood, I can confirm this issue is fixed. So appart from the TensorFlow workaround, everything works fine I think.
